subsys/emul/espi/espi_emul_host.c

```diff
@@ -0,0 +1,388 @@
+	 */
+	struct espi_emul_vw_data vw_state[NUMBER_OF_VWIRES];
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	/** ACPI Shared memory. */
+	uint8_t shm_acpi_mmap[CONFIG_ESPI_EMUL_HOST_ACPI_SHM_REGION_SIZE];
+#endif
+#ifdef CONFIG_ESPI_PERIPHERAL_CUSTOM_OPCODE
+	/** Host commands shared memory. */
+	uint8_t shm_host_cmd[CONFIG_ESPI_EMUL_HOST_CMD_SHM_REGION_SIZE];
+#endif
+};
+
+/** Static configuration for the emulator */
+struct espi_host_emul_cfg {
+	/** Label of the eSPI bus this emulator connects to */
+	const char *espi_label;
+	/** Label of the emulated AP*/
+	const char *label;
+	/* eSPI chip-select of the emulated device */
+	uint16_t chipsel;
+};
+
+/* Helpers */
+
+/**
+ * Initialize the state of virtual wires to default based on
+ * the vw_state_default array.
+ *
+ * @param data Host emulator data with the vwire array
+ */
+static void emul_host_init_vw_state(struct espi_host_emul_data *data)
+{
+	unsigned int i;
+
+	for (i = 0; i < NUMBER_OF_VWIRES; i++) {
+		data->vw_state[i] = vw_state_default[i];
+	}
+}
+
+/**
+ * Find a virtual wire in the array placed in the host data.
+ *
+ * @param data Host emulator data with the vwire array
+ * @param vw Virtual wire signal to be found
+ * @return index in the array
+ * @return -1 if not found
+ */
+static int emul_host_find_index(struct espi_host_emul_data *data,
+				enum espi_vwire_signal vw)
+{
+	int idx;
+
+	for (idx = 0; idx < NUMBER_OF_VWIRES; idx++) {
+		if (data->vw_state[idx].sig == vw) {
+			return idx;
+		}
+	}
+
+	return -1;
+}
+
+/* API */
+
+static int host_espi_api_config(const struct espi_emul *dev, struct espi_cfg *cfg)
+{
+	return 0;
+}
+
+static bool host_espi_api_get_channel_status(const struct espi_emul *dev,
+					    enum espi_channel ch)
+{
+	return ch == ESPI_CHANNEL_VWIRE;
+}
+
+/* Logical Channel 0 APIs */
+static int host_espi_api_read_request(const struct espi_emul *dev,
+				     struct espi_request_packet *req)
+{
+	return -EIO;
+}
+
+static int host_espi_api_write_request(const struct espi_emul *dev,
+				      struct espi_request_packet *req)
+{
+	return -EIO;
+}
+
+static int host_espi_api_lpc_read_request(const struct espi_emul *dev,
+					 enum lpc_peripheral_opcode op,
+					 uint32_t *value)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+
+	switch (op) {
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	case EACPI_GET_SHARED_MEMORY:
+		*value = (uint32_t)data->shm_acpi_mmap;
+		break;
+#endif
+#ifdef CONFIG_ESPI_PERIPHERAL_CUSTOM_OPCODE
+	case ECUSTOM_HOST_CMD_GET_PARAM_MEMORY:
+		*value = (uint32_t)data->shm_host_cmd;
+		break;
+#endif
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int host_espi_api_lpc_write_request(const struct espi_emul *dev,
+					  enum lpc_peripheral_opcode op,
+					  uint32_t *value)
+{
+#ifdef CONFIG_ESPI_PERIPHERAL_CUSTOM_OPCODE
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
```

Comment:
@aaronemassey Does this affect your emulator API change?

Reply:
It will, but it'll only be minor changes to function signatures on my part.